### PR TITLE
created tests for FruitShop

### PR DESCRIPTION
**Pull request by maaarynet**

### CODE REVIEW

File: src/test/java/core/basesyntax/filewriter/CsvFileWriterImplTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class CsvFileWriterImplTest {
+    private static final String EXPECTED_CONTENT = "fruit,quantity\nbanana,20";
+    private static final String INVALID_PATH = "/invalid/path/test.csv";
+    private static final String TEMP_FILE_PREFIX = "test_output";
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/filewriter/CsvFileWriterImplTest.java:16]: you should create test files in the test resources directory

File: src/test/java/core/basesyntax/operation/BalanceOperationTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class BalanceOperationTest {
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/operation/BalanceOperationTest.java:11]: you should clear storage after testing. fix it everywhere.

File: src/test/java/core/basesyntax/reportgenerator/ReportGeneratorImplTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    void getReport_withBalanceData_returnsFormattedReport() {
+        List<FruitTransaction> transactions = List.of(BALANCE_TRANSACTION);
+        shopService.process(transactions);
+
+        String report = reportGenerator.getReport();
+        String expectedLine = String.format(EXPECTED_LINE_FORMAT, TEST_FRUIT, TEST_QUANTITY);
+
+        assertTrue(report.contains(HEADER), "Report should contain header");
+        assertTrue(report.contains(expectedLine),
+                "Report should contain correct fruit and quantity");
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/reportgenerator/ReportGeneratorImplTest.java:48]: just compare the expected and actual reports. Can you add more test cases? For example with empty storage

File: src/test/java/core/basesyntax/shopservice/ShopServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import core.basesyntax.operation.OperationStrategyImpl;
+import core.basesyntax.operation.PurchaseOperation;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ShopServiceImplTest {
```
Comment (liliia-ponomarenko) [inline src/test/java/core/basesyntax/shopservice/ShopServiceImplTest.java:18]: the same here. add more than one test case.

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  improve your solution ;)

**Review by JJJazl — APPROVED** [review]
Great! Let's move on)
